src/huggingface_hub/inference/_templating.py

```diff
@@ -0,0 +1,105 @@
+from functools import lru_cache
+from typing import Callable, Dict, List, Optional, Union
+
+from ..utils import HfHubHTTPError, RepositoryNotFoundError, is_minijinja_available
+
+
+class TemplateError(Exception):
+    """Any error raised while trying to fetch or render a chat template."""
+
+
+def _import_minijinja():
+    if not is_minijinja_available():
+        raise ImportError("Cannot render template. Please install minijinja using `pip install minijinja`.")
+    import minijinja  # noqa: F401
+
+    return minijinja
+
+
+def render_chat_prompt(
+    *,
+    model_id: str,
+    messages: List[Dict[str, str]],
+    token: Union[str, bool, None] = None,
+    add_generation_prompt: bool = True,
+    **kwargs,
+) -> str:
+    """Render a chat prompt using a model's chat template.
+
+    Args:
+        model_id (`str`):
+            The model id.
+        messages (`List[Dict[str, str]]`):
+            The list of messages to render.
+        token (`str` or `bool`, *optional*):
+            Hugging Face token. Will default to the locally saved token if not provided.
+
+    Returns:
+        `str`: The rendered chat prompt.
+
+    Raises:
+        `TemplateError`: If there's any issue while fetching, compiling or rendering the chat template.
+    """
+    minijinja = _import_minijinja()
+    template = _fetch_and_compile_template(model_id=model_id, token=token)
+
+    try:
+        return template(messages=messages, add_generation_prompt=add_generation_prompt, **kwargs)
+    except minijinja.TemplateError as e:
+        raise TemplateError(f"Error while trying to render chat prompt for model '{model_id}': {e}") from e
+
+
+@lru_cache  # TODO: lru_cache for raised exceptions
```

Comment:
Does that mean that if a remote repository updates its chat template, then a runtime restart will be necessary to get the updated template?

Reply:
(probably nitpick)

Reply:
> Does that mean that if a remote repository updates its chat template, then a runtime restart will be necessary to get the updated template?

Yes indeed, that's the case.
Not worth improving IMO. And in any case, this is "only" for models not served with TGI so not the common use case.